src/theory/ff/theory_ff.h

```diff
@@ -63,16 +65,8 @@ class TheoryFiniteFields : public Theory
   /** Collect model values in m based on the relevant terms given by termSet */
   bool collectModelValues(TheoryModel* m,
                           const std::set<Node>& termSet) override;
-  void computeCareGraph() override;
-  TrustNode explain(TNode) override;
```

Comment:
Have you tried finite fields in the presence of other theories, e.g. UF over finite fields?  The `explain` method is typically key to this.  I think you would keep `explain` and implement it by forwarding to TheoryInferenceManager, see e.g. https://github.com/cvc5/cvc5/blob/main/src/theory/strings/theory_strings.cpp#L191.

Perhaps it would be good to have an example combining FF with UF where FF propagates an equality over an equality between shared terms.  Something like (use `--simplification=none`):

```
(declare-fun f (FF) FF)
(declare-fun a () FF)
(declare-fun b () FF)
(declare-fun c () FF)
(assert (and (= a b) (= b c) (not (= (f a) (f c))))
(check-sat)
```
where FF is a finite field type.

Reply:
Is `explain` necessary for correctness or only performance? I'm interested either way, of course.

I just pushed a commit which fixed up FF theory combination by no longer assuming that EQs are preRegistered.

I wrote a bunch of QF_UFFF regression tests, which revealed a few other bugs. I fixed them too. However, those regressions seem to work without `explain`.

Reply:
> If your solver is not reaching that line, then perhaps propagation has not been fully set up for your theory (and hence it will suffer performance-wise in combined theories).

Yeah, right now we don't propagate at all. Should I also be delegating some propagation to the TheoryInferenceManager?

Reply:
Update: with Andy's help, we now have a regression that triggers this.